Bigtable/src/Connection/Grpc.php

```diff
@@ -159,79 +181,141 @@ public function deleteInstance(array $args)
      */
     public function createCluster(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $parent = $this->pluck('parent', $args);
+        $response = $this->send([$this->bigtableInstanceAdminClient, 'createCluster'], [
+            $parent,
+            $this->pluck('clusterId', $args),
+            $this->clusterObject(
+                $this->pluck('cluster', $args)
+            ),
+            $this->addResourcePrefixHeader($args, $parent)
+        ]);
+
+        return $this->operationToArray(
+            $response,
+            $this->serializer,
+            $this->lroResponseMappers
+        );
     }
 
     /**
      * @param array $args
      */
     public function getCluster(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $name = $this->pluck('name', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'getCluster'], [
+            $name,
+            $this->addResourcePrefixHeader($args, $name)
+        ]);
     }
 
     /**
      * @param array $args
      */
     public function listClusters(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $parent = $this->pluck('parent', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'listClusters'], [
+            $parent,
+            $this->addResourcePrefixHeader($args, $parent)
+        ]);
     }
 
     /**
      * @param array $args
      */
     public function updateCluster(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $name = $this->pluck('name', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'updateCluster'], [
+            $name,
+            $this->pluck('location', $args),
```

Comment:
we can not update location of cluster once created, its creationOnly.

Reply:
Ack. This might be something we need to touch on in the GAPIC generation then as it is a required field: https://github.com/GoogleCloudPlatform/google-cloud-php/blob/bigtable/Bigtable/src/Admin/V2/Gapic/BigtableInstanceAdminGapicClient.php#L1126

Reply:
The field is ignored on the server-side, so just pass in empty from the user-facing layer.

Reply:
That will work, but for users of the GAPIC itself this is very confusing.